examples/typescript/src/App.tsx

```diff
@@ -0,0 +1,35 @@
+import React, {useMemo} from 'react';
+import {locations} from "@contentful/app-sdk";
+import ConfigScreen from "./locations/ConfigScreen";
+import Field from "./locations/Field";
+import EntryEditor from "./locations/EntryEditor";
+import Dialog from "./locations/Dialog";
+import Sidebar from "./locations/Sidebar";
+import Page from "./locations/Page";
+import {useSDK} from "@contentful/react-apps-toolkit";
+
+const ComponentLocationSettings = {
+  [locations.LOCATION_APP_CONFIG]: <ConfigScreen />,
+  [locations.LOCATION_ENTRY_FIELD]: <Field />,
+  [locations.LOCATION_ENTRY_EDITOR]: <EntryEditor />,
+  [locations.LOCATION_DIALOG]: <Dialog />,
+  [locations.LOCATION_ENTRY_SIDEBAR]: <Sidebar />,
+  [locations.LOCATION_PAGE]: <Page />,
+}
+
+const App = () => {
+  const sdk = useSDK()
+
+  const RenderedComponent = useMemo(() => {
+    for (const [location, component] of Object.entries(ComponentLocationSettings)) {
+      if (sdk.location.is(location)) {
+        return component;
+      }
+    }
+  }, [sdk.location])
```

Comment:
Let's make sure we can get rid of this loop in the future - then everything is shiny :)